### PR TITLE
MRG: CircleCI 2.0

### LINKED ISSUE #349: BUG: CircleCI 1.0 is retiring
... in August. So we need to migrate to 2.0. This *should* be pretty straightforward given that I've already done it for MNE-Python.

@drammock you can do it if you want to brush up on your CI skills, otherwise I'll do it.

### PR DESCRIPTION
**Pull request by larsoner**
Closes #349.

### CODE REVIEW

**Comment by larsoner** [comment]
@drammock ready for review/merge from my end

**Comment by drammock** [comment]
I can review this on monday when I'm back in the office

**Comment by drammock** [comment]
```
(three) [~/B/expyfun] $ git chb testcircle eric/circle 
Branch testcircle set up to track remote branch circle from eric.
Switched to a new branch 'testcircle'
(three) [~/B/expyfun] $ make test
rm -rf build
find . -name "*.pyc" | xargs rm -f
find . -name "*.so" | xargs rm -f
find . -name "*.pyd" | xargs rm -f
rm -f tags
find . -name "__pycache__" | xargs rm -rf
rm -f .coverage
pytest expyfun
usage: pytest [options] [file_or_dir] [file_or_dir] [...]
pytest: error: unrecognized arguments: --cov=expyfun --cov-report=
  inifile: /home/drmccloy/Builds/expyfun/setup.cfg
  rootdir: /home/drmccloy/Builds/expyfun
Makefile:44: recipe for target 'nosetests' failed
make: *** [nosetests] Error 2
```

**Comment by larsoner** [comment]
install `pytest-cov`

**Comment by drammock** [comment]
is my version of pytest too old? (maybe we should check for that?)

**Comment by larsoner** [comment]
No you need both `pytest` and `pytest-cov` packages

**Comment by drammock** [comment]
test pass in my python3 env, but in my (rarely-used) py2 environment I'm getting an enormous traceback culminating in `RuntimeError: maximum recursion depth exceeded`.  Should I ignore it or try to track it down?

**Comment by drammock** [comment]
found problem.  Old pip-installed version of `py` was overriding newer conda-installed version of `py`.  *sigh*

**Comment by larsoner** [comment]
https://xkcd.com/1987/

**Comment by larsoner** [comment]
Oh wow, never seen that one before.

File: expyfun/tests/test_version.py [hunk 1]
```diff
@@ -3,58 +3,60 @@
...
                 line1 = fid.readline().strip()
             got_version = line1.split(' = ')[1][-8:-1]
-            ex = want_version if want_version != 'cae6bc3' else '.dev0+c'
-            assert_equal(got_version, ex,
-                         msg='File {0}: {1} != {2}'.format(got_fname,
-                                                           got_version, ex))
+            ex = want_version
+            if want_version == 'cae6bc3':
+                ex = (ex, '.dev0+c')
+            assert got_version in ex, got_fname
```
Comment (drammock) [inline expyfun/tests/test_version.py]: this line I don't get.  why is `got_fname` here?

Comment (drammock) [reply expyfun/tests/test_version.py]: (i.e., why would the `got_version` ever equal the file path, and why would we want to allow that?)

Comment (larsoner) [reply expyfun/tests/test_version.py]: In `assert` statements what follows the comma is a message (it is not creating a tuple), e.g.:
```
>>> assert 'dan' in 'cool', 'he is not'
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
AssertionError: he is not
```
In this case it's useful to have the filename printed, especially when using `pytest --pdb`.

Comment (drammock) [reply expyfun/tests/test_version.py]: I've been pwned

**Comment by drammock** [comment]
thanks @larsoner

**Pull request merged**